src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,289 @@
+# pylint: disable=wrong-import-order
+
+from __future__ import annotations
+
+from typing import Any
+import numpy as np
+import pandas as pd
+import regex
+from unittest import mock
+
+from dsp_tools.models.exceptions import UserError
+
+languages = ["en", "de", "fr", "it", "rm"]
+
+
+def read_excel_file(excel_filename: str) -> pd.DataFrame:
+    """
+    This function reads an Excel file, if there is a ValueError then it patches the openpyxl part that creates the
+    error and opens it with that patch. It cleans and then returns the pd.DataFrame.
+
+    Args:
+        excel_filename: The name of the Excel file
+
+    Returns:
+        A pd.DataFrame
+    """
+    # load file
+    try:
+        read_df: pd.DataFrame = pd.read_excel(excel_filename)
+    except ValueError:
+        # Pandas relies on openpyxl to parse XLSX files.
+        # A strange behaviour of openpyxl prevents pandas from opening files with some formatting properties
+        # (unclear which formatting properties exactly).
+        # Apparently, the excel2json test files have one of the unsupported formatting properties.
+        # Credits: https://stackoverflow.com/a/70537454/14414188
+        with mock.patch("openpyxl.styles.fonts.Font.family.max", new=100):
+            read_df = pd.read_excel(excel_filename)
+    read_df = clean_data_frame(unclean_df=read_df)
```

Comment:
There is a rule in Software Architecture that says "Do one thing, and do it well". It is confusing that the function `read_excel_file()` reads the Excel file and cleans it. Better: take the cleaning out, and call it separately, like:

```
# properties.py:
property_df = utl.read_excel_file(excel_filename=excelfile)
property_df = clean_data_frame(unclean_df=property_df)
```

Reply:
I like to put it in the read function, because I always want to clean the excel file... So it makes it easier. Do you think it would make sense instead to write ead_clean_excel_file? Or should we make a function that only reads it and a function that calls those two functions?

Reply:
I think renaming this function to `read_and_clean_excel_file()` would be fine. Then the caller knows what he will get, and it's not any more a blackbox that does unexpected things.